### PR TITLE
feat: Add snowflake_password_policies data source

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
## Summary
- Add `snowflake_password_policies` data source with `like`, `in`, `limit` filtering and optional `with_describe` toggle
- Register as a preview feature (`snowflake_password_policies_datasource`)
- Add acceptance tests covering basic usage, all filtering axes, describe toggle, and empty-in validation
- Add migration guide entry, documentation, and examples

### CODE REVIEW

File: pkg/datasources/password_policies.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+var passwordPoliciesSchema = map[string]*schema.Schema{
+	"with_describe": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     true,
+		Description: "Runs DESC PASSWORD POLICY for each password policy returned by SHOW PASSWORD POLICIES. The output of describe is saved to the describe_output field. By default this value is set to true.",
+	},
+	"like":  likeSchema,
+	"in":    inSchema,
+	"limit": limitFromSchema,
```
Comment (sfc-gh-kwasilewski) [inline pkg/datasources/password_policies.go]: It's missing `starts_with` and `on` attributes

**Review by sfc-gh-kwasilewski — CHANGES_REQUESTED** [review]

File: pkg/datasources/password_policies.go (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply pkg/datasources/password_policies.go]: Added. Also, added support for BUILT-IN policy (the same case as in authentication policies).

File: pkg/sdk/testint/password_policy_integration_test.go [hunk 2]
```diff
@@ -267,36 +267,36 @@ func TestInt_PasswordPolicies(t *testing.T) {
 		t.Run("like", func(t *testing.T) {
```
Comment (sfc-gh-kwasilewski) [inline pkg/sdk/testint/password_policy_integration_test.go:267]: `starts_with`, `on_user`, and `on_account` are not tested here

**Review by sfc-gh-kwasilewski — APPROVED** [review]

File: pkg/sdk/testint/password_policy_integration_test.go (hunk 2 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/password_policy_integration_test.go:267]: Hmm, yeah. We don't always test each combination, but I can add a few tests.
